### DISCUSSION TITLE
Inclusion of Vector Database in PocketBase's Roadmap

### DISCUSSION THREAD

**Discussion by rossanodr** [pocketbase/pocketbase#2990] [Q&A] [answered]
Hello PocketBase team! I'm curious to know if you have any plans to include a vector database in the future. It would be great to hear if vector database support is on the roadmap. Thanks!

**Comment by ganigeorgiev** [comment, marked as answer]
No sorry, although there some existing SQLite extensions that could help with this, there are no plans implementing it since it will be hard to make it portable and at least for time being it remains out the scope for v1.

**Comment by de1mat** [comment]
Came here for the same reason. Wanting to store vectors in PocketBase. Is it easy enough to get custom SQLite extensions working if using PocketBase as a framework? If so, any guidelines or docs around this to get started?

**Reply by ganigeorgiev** [reply]
> Is it easy enough to get custom SQLite extensions working if using PocketBase as a framework

Custom C extensions can be registered only when using the CGO db driver but we currently don't expose directly the implementation (see [`db_cgo.go`](https://github.com/pocketbase/pocketbase/blob/master/core/db_cgo.go)).

But even if you manage to setup the extension, I'm not sure what would be the benefits since you may not be able to interact with it using the PocketBase web APIs.

PocketBase is not the best tool for such use cases. If you want to use a BaaS for your project, you may want to check Supabase since I think they have integration with pgvector.

**Reply by de1mat** [reply]
Thanks Gani. Yes Supabase do support vectors but I was hoping to use PocketBase for an internal project. I haven't spent much time studying the internals of PocketBase yet but if it is doable working with it as a framework and doing custom builds I would be keen to examine it further.

**Comment by vlasky** [comment]
I strongly encourage you to include a vector database on the roadmap. Lots of people are now building innovative LLM/ChatGPT apps with Retrieval-Augmented Generation (RAG) and having a vector database capability in PocketBase would allow it to be a complete solution.

**Reply by kittykatattack** [reply]
Yup, you will Rule The World with a vector database!

**Comment by greendesertsnow** [comment]
Why not just run a binary quantized Qdrant server? Use specialized tools for specialized works. 
It's fast, it's efficient...
Pgvector is not a good invention either. Supabase is has a terrible DX, if you don't like writing your rules in pg language.

**Comment by petermakeswebsites** [comment]
I would like a vector database as well!

[Plugin for SQLite](https://alexgarcia.xyz/blog/2024/sqlite-vec-stable-release/index.html)
